src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
They're "natively" supported, just not enabled OOTB. Keeping track of down nodes may be enough of a perf benefit to make it an option to users. Only benchmarking will tell.